src/Compilers/CSharp/Test/Emit3/Attributes/AttributeTests_StructLayout.cs

```diff
@@ -19,6 +19,122 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+                    public AttributeUsageAttribute(AttributeTargets validOn, bool allowMultiple, bool inherited) {}
+                    public bool Inherited { get; set; }
+                    public bool AllowMultiple { get; set; }
+                }
+
+                public struct UInt16
+                {
+                    private ushort m_value;
+                }
+
+                public struct Int32
+                {
+                    private int m_value;
+                }
+
+                public struct Boolean
+                {
+                    private bool m_value;
+                }
+
+                public struct Single
+                {
+                    private float m_value;
+                }
+            }
+
+            namespace System.Runtime.InteropServices
+            {
+                [AttributeUsage(AttributeTargets.Struct | AttributeTargets.Class)]
+                public sealed class StructLayoutAttribute(LayoutKind kind): Attribute
+                {
+                    public StructLayoutAttribute(ushort kind) : this((LayoutKind)kind){}
+                }
+            
+                public enum LayoutKind
+                {
+                    Sequential = 0,
+                    Extended = 1,
+                    Explicit = 2, 
+                    Auto = 3
+                }
+            
+                [AttributeUsage(AttributeTargets.Field, Inherited = false)]
+                public sealed class FieldOffsetAttribute : Attribute
+                {
+                    public FieldOffsetAttribute(int offset)
+                    {
+                    }
+                    public int Value { get { return 0; } }
+                }
+            
+                [AttributeUsage(AttributeTargets.Struct)]
+                #pragma warning disable CS9113
+                public sealed class ExtendedLayoutAttribute(ExtendedLayoutKind kind): Attribute;
+                #pragma warning restore CS9113
+            
+                public enum ExtendedLayoutKind
+                {
+                    CStruct,
+                    CUnion
+                }
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                [AttributeUsage(AttributeTargets.Struct)]
+                #pragma warning disable CS9113
+                public sealed class InlineArrayAttribute(int repeat): Attribute;
+                #pragma warning restore CS9113
+            }
+            """;
+
+        private static readonly MetadataReference ExtendedLayoutAttributeMinimalCoreLibrary = CSharpCompilation.Create("System.Private.CoreLib",
+            syntaxTrees:
+            [
+                CSharpSyntaxTree.ParseText(
+                    ExtendedLayoutMinimalCoreLibrary,
+                    options: new CSharpParseOptions(LanguageVersion.Preview))
+            ],
+            options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)).ToMetadataReference();
```

Comment:
Note, this is from memory on github so +/- typos.

```suggestion
        private static readonly MetadataReference ExtendedLayoutAttributeMinimalCoreLibrary = CreateEmptyCompilation(ExtendedLayoutMinimalCoreLibrary, parseOptions: TestOptions.RegularNext).ToMetadataReference();
```

Reply:
Also, why did you specify langversion preview? I wouldn't expect any special parse options to be necessary, actually.